src/Features/CSharp/Portable/Completion/CompletionProviders/DeclarationNameCompletionProvider.cs

```diff
@@ -111,7 +111,7 @@ private bool IsValidType(ITypeSymbol type)
                 return false;
             }
 
-            if (type.IsErrorType() && (type.Name == "var" || type.Name == string.Empty))
```

Comment:
I think it would be better to enforce [CA1820](https://docs.microsoft.com/visualstudio/code-quality/ca1820): https://github.com/dotnet/roslyn/blob/470b49b3dd37077dbd338e560e96f5f9762998e7/eng/config/rulesets/Shipping.ruleset#L53

Reply:
@mavasani, I didn't update the usage of Length in all cases.
There are cases where it's not clear if the string could be null or not. Using `.Length` in such cases will have different behavior. For example:

```csharp
string s = null;
s == string.Empty; // false
s.Length == 0; // throws an exception.
```